datafusion/src/sql/planner.rs

```diff
@@ -2749,14 +2772,139 @@ mod tests {
         );
     }
 
+    /// psql result
```

Comment:
I don't think it really matters, but the `max`, `min` and `avg` window functions don't actually depend on order (and so in theory all of the sorts here could be optimized away.

Reply:
they do. with `order by` they compute accumulative sum/avg/max/min, not a full partition one.